packages/flutter/lib/src/widgets/editable_text.dart

```diff
@@ -836,13 +860,19 @@ class EditableTextState extends State<EditableText> with AutomaticKeepAliveClien
       widget.controller.addListener(_didChangeTextEditingValue);
       _updateRemoteEditingValueIfNeeded();
     }
+    if (widget.controller.selection != oldWidget.controller.selection) {
+      _selectionOverlay?.update(_value);
```

Comment:
after play a little bit more, I figure we do not update the overlay when selection change during didupdatewidget.
I filed https://github.com/flutter/flutter/issues/32152 to demonstrate this issue.
added test 'Sawping controllers should update selection'

Reply:
I cannot separate this into different pr, since the readonly requires this issue to fix to work properly.